api/src/test/java/org/openmrs/messagesource/MessageSourceServiceTest.java

```diff
@@ -0,0 +1,47 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.messagesource;
+
+import org.junit.Assert;
+import org.junit.Test;
+import org.openmrs.api.context.Context;
+import org.openmrs.messagesource.impl.MessageSourceServiceImpl;
+import org.openmrs.test.BaseContextSensitiveTest;
+import org.openmrs.test.Verifies;
+import org.springframework.context.MessageSourceResolvable;
+import org.springframework.validation.MapBindingResult;
+
+import java.util.HashMap;
+import java.util.Locale;
+
+/**
+ *
+ */
+public class MessageSourceServiceTest extends BaseContextSensitiveTest {
+
+    /**
+     * MessageSourceServiceImpl.getMessage()should return last error code if no localization found
+     *
+     * @see MessageSourceServiceImpl#getMessage(MessageSourceResolvable resolvable, Locale locale)
+     */
+    @Test
+    @Verifies(value = "getMessage_shouldReturnTheLastErrorCodeIfnoLocalizationIsFound", method = "getMessage(MessageSourceResolvable resolvable, Locale locale)")
+    public void getMessage_shouldReturnTheLastErrorCodeIfnoLocalizationIsFound() throws Exception {
+        MapBindingResult errors = new MapBindingResult(new HashMap<String, Object>(), "request");
+        errors.rejectValue("myField", "myErrorCode");
+        MessageSourceResolvable fieldError = errors.getFieldError("myField");
+        Assert.assertEquals(3, fieldError.getCodes().length);
+        Assert.assertEquals("myErrorCode.request.myField", fieldError.getCodes()[0]);
+        Assert.assertEquals("myErrorCode.myField", fieldError.getCodes()[1]);
+        Assert.assertEquals("myErrorCode", fieldError.getCodes()[2]);
+        Assert.assertNotEquals("myErrorCode.request.myField", Context.getMessageSourceService().getMessage(fieldError, Context.getLocale()));
```

Comment:
This is unnecessary. The next line asserts what this value _should_ be. You don't need to also assert what it should _not_ be.

(In fact, if this test were ever to fail, it would be confusing to whoever looks at it because they'd be seeing the failure on this line, i.e. should _not_ be xyz, instead of the next line saying what the value _should_ be.)